src/groups/rotation_translation_action.jl

```diff
@@ -0,0 +1,310 @@
+    a::ArrayPartition,
+    p,
+)
+    return a.x[2] \ (p - a.x[1])
+end
+function apply(
+    ::RotationTranslationActionOnVector{RightForwardAction},
+    a::SpecialEuclideanIdentity,
+    p,
+)
+    return p
+end
+
+function apply!(
+    ::RotationTranslationActionOnVector{LeftForwardAction},
+    q,
+    a::ArrayPartition,
+    p,
+)
+    mul!(q, a.x[2], p)
+    q .+= a.x[1]
+    return q
+end
+function apply!(
+    ::RotationTranslationActionOnVector{LeftForwardAction},
+    q,
+    a::SpecialEuclideanIdentity,
+    p,
+)
+    copyto!(q, p)
+    return q
+end
+
+function inverse_apply(
+    ::RotationTranslationActionOnVector{LeftForwardAction},
+    a::ArrayPartition,
+    p,
+)
+    return a.x[2] \ (p - a.x[1])
+end
+function inverse_apply(
+    ::RotationTranslationActionOnVector{RightForwardAction},
+    a::ArrayPartition,
+    p,
+)
+    return a.x[2] * p + a.x[1]
+end
+
+function apply_diff(
+    ::RotationTranslationActionOnVector{LeftForwardAction},
+    a::ArrayPartition,
+    p,
+    X,
+)
+    return a.x[2] * X
+end
+function apply_diff(
+    ::RotationTranslationActionOnVector{LeftForwardAction},
+    ::SpecialEuclideanIdentity,
+    p,
+    X,
+)
+    return X
+end
+function apply_diff(
+    ::RotationTranslationActionOnVector{RightForwardAction},
+    a::ArrayPartition,
+    p,
+    X,
+)
+    return a.x[2] \ X
+end
+function apply_diff(
+    ::RotationTranslationActionOnVector{RightForwardAction},
+    a::SpecialEuclideanIdentity,
+    p,
+    X,
+)
+    return X
+end
+
+function apply_diff!(
+    ::RotationTranslationActionOnVector{LeftForwardAction},
+    Y,
+    a::ArrayPartition,
+    p,
+    X,
+)
+    return mul!(Y, a.x[2], X)
+end
+function apply_diff!(
+    ::RotationTranslationActionOnVector{LeftForwardAction},
+    Y,
+    a::SpecialEuclideanIdentity,
+    p,
+    X,
+)
+    return copyto!(Y, X)
+end
+function apply_diff!(
+    ::RotationTranslationActionOnVector{RightForwardAction},
+    Y,
+    a::ArrayPartition,
+    p,
+    X,
+)
+    Y .= a.x[2] \ X
+    return Y
+end
+function apply_diff!(
+    ::RotationTranslationActionOnVector{RightForwardAction},
+    Y,
+    a::SpecialEuclideanIdentity,
+    p,
+    X,
+)
+    return copyto!(Y, X)
+end
+
+function apply_diff_group(
+    ::RotationTranslationActionOnVector{LeftForwardAction},
+    ::SpecialEuclideanIdentity,
+    X,
+    p,
+)
+    return X.x[2] * p
+end
+
+function apply_diff_group!(
+    ::RotationTranslationActionOnVector{LeftForwardAction},
+    Y,
+    ::SpecialEuclideanIdentity,
+    X::ArrayPartition,
+    p,
+)
+    Y .= X.x[2] * p
+    return Y
+end
+
+function inverse_apply_diff(
+    ::RotationTranslationActionOnVector{LeftForwardAction},
+    a::ArrayPartition,
+    p,
+    X,
+)
+    return a.x[2] \ X
+end
+function inverse_apply_diff(
+    ::RotationTranslationActionOnVector{RightForwardAction},
+    a::ArrayPartition,
+    p,
+    X,
+)
+    return a.x[2] * X
+end
+
+###
+
+@doc raw"""
```

Comment:
Could you motivate how. this is used later? I would have thought this is more like a product of actions, so that I am not sure how much this type is needed.

Reply:
We don't currently have a generic "product" of actions, and I'm not sure how much it would be needed. We can always refactor it later when the appropriate product (I can think of many different constructions that can be thought of as a product of actions) is introduced.

Reply:
Ok, this sounds fair.